### ISSUE TITLE
(To be discussed) 'yunohost' in the CLI could be slightly smarter and implicitly do 'sudo' if ran from admin

### ISSUE THREAD

**Issue by alexAubin** [YunoHost/issues#1565]
Since we now encourage / force people to connect as 'admin' instead of 'root', I regularly see people "forgetting" to put the sudo in front of every commands ...

Yunohost could do some magic to escalate to root automatically. Though I dunno, maybe it would create more confusion about having to sudo for other commands but not for yunohost ...

**Comment by lhp22** [comment]
That could be an option disabled by default, but i think that would be usefull.

**Comment by Aeris1One** [comment]
As Yunohost replace root with admin, I think admin should behave like root.

**Comment by autra** [comment]
> I regularly see people "forgetting" to put the sudo in front of every commands ...

For me it's a feature. The good thing about `sudo` is to force people to mean it when they want to go root (and not be root all the time). I'm even tempted to put back the password prompt on my server.

Yunohost cli could sudo itself, but then it should ask for the password IMHO. The idea is always to have a difference between root and non root actions,WDYT?

> As Yunohost replace root with admin, I think admin should behave like root.

In this case, why having an admin user instead of root directly?

**Comment by alexAubin** [comment]
Yeah, I kind of agree with you @autra , my initial idea was based on `apt`'s behavior which now tends to automatically sudo. Same for systemctl I believe (maybe not exactly sudo but something similar..)

>In this case, why having an admin user instead of root directly?

I think historically this was because we didn't realize we could auth on LDAP just by being root ... Back in the days, you had to provide that admin password everytime you ran something like 'yunohost user create' even when being logged as root lol ... But nowadays it's less clear indeed (except for that security practice of not SSH-ing as root and avoiding to be root all the time...).

**Comment by autra** [comment]
> `apt`'s behavior which now tends to automatically sudo. Same for systemctl I believe (maybe not exactly sudo but something similar..)

yeah, that's true. I'm not sure I like that, as I think the people who made that change didn't really think about sudo without password. Or they thought "if they remove the password it's their problem".

Maybe ynh could exclude systemctl and and ynh cli from the no password rule as a safety guard?

**Comment by Josue-T** [comment]
For me it could be good to give the possibility to run yunohost command without sudo. My point of view is that when you are logged as root you know that can do everything on your server and you know that some action could break everything. Yunohost command a generally secure, I don't think that it's possible to break your server with yunohost command except if your install a very bad app...

So for me having the concept to just say that you can run yunohost command as admin give the feeling to the user that there are a difference between running a yunohost command and running any other command as root. For me this is an interesting feature

**Comment by autra** [comment]
>  Yunohost command a generally secure, I don't think that it's possible to break your server with yunohost command

Yes it is :-) I've already broken an app by upgrading system packages and the app in the wrong order (the app was updated because of a change in ynh, I guess about php, and you were supposed to upgrade the system before the app). And I guess a lot of people do break ynh from time to time (seeing the number of messages in the support room). Maybe it's their fault, maybe not...

Maybe a good middle ground would be to allow every non-dangerous command like creating backups, getting various info etc... to be executed without password, and only ask password, or a confirmation, for every command that modifies the system? That way, this would put emphasis on what is *really* important?

**Comment by Aeris1One** [comment]
Agree with both @Josue-T and @autra, when you're logged with root/admin you should know what you're doing but a confirmation when doing something that edits the system would be great, but maybe not asking for password, maybe more like apt:
`Are you sure you want to do that ? [y/N]`
As I usually use randomly generated passwords on my Yunohost instances, I want to enter it only one time (when logging in).

**Comment by alexAubin** [comment]
Closing because 5 years old, doesn't seem to be a huge deal
